packages/core/src/Serialization/TransactionWitnessSet/TransactionWitnessSet.ts

```diff
@@ -111,15 +123,38 @@ export class TransactionWitnessSet {
 
     if (this.#redeemers && this.#redeemers.length > 0) {
       writer.writeInt(5n);
-      writer.writeStartArray(this.#redeemers.length);
 
-      for (const data of this.#redeemers) {
-        writer.writeEncodedValue(Buffer.from(data.toCbor(), 'hex'));
+      if (CborWriter.useConwaySerialization) {
```

Comment:
Similar to the Set, we could create Reedemeer**s** class, which knowns how to serialize/deserialize the redeemers into an array or a map depending on the era